crates/topos-tce-broadcast/src/double_echo/mod.rs

```diff
@@ -25,6 +25,10 @@ pub struct DoubleEcho {
     task_manager_message_sender: mpsc::Sender<DoubleEchoCommand>,
     /// The overview of the network, which holds echo and ready subscriptions and the network size
     pub subscriptions: SubscriptionsView,
+    /// Public ETH address
+    pub authority_id: AuthorityId,
+    /// Known and approved validators in the network
+    pub validators: Vec<String>,
```

Comment:
I think that you want this set of `AuthorityId` in the sampler directly so that your different HashSet in `subscriptions` are directly fed with `AuthorityId`s no?

Reply:
I am not sure about this part. Since we don't want to touch the `libp2p` part where the discovery is done through `PeerId`s. I have to go through thoroughly and check where exactly we need `PeerId`s and where we use `AuthorityId`.